srcalc/src/main/java/gov/va/med/srcalc/web/view/ReferenceInfoAdder.java

```diff
@@ -0,0 +1,81 @@
+package gov.va.med.srcalc.web.view;
+
+import gov.va.med.srcalc.domain.HealthFactor;
+import gov.va.med.srcalc.domain.Patient;
+import gov.va.med.srcalc.domain.model.Variable;
+import gov.va.med.srcalc.domain.model.VariableGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+/**
+ * <p>A class used to add reference information to existing PopulatedDisplayGroups and create
+ * PopulatedDisplayGroups that do not exist, in order to store reference information to display
+ * for the user.</p>
+ * 
+ * <p>Per Effective Java Item 17, this class is marked final because it was not
+ * designed for inheritance.</p>
+ */
+public final class ReferenceInfoAdder
+{
+    /**
+     * Adds reference information for all of the given {@link PopulatedDisplayGroup}s and adds any groups
+     * that are missing in order to include their reference information as well.
+     * @param map a map where they key is a VariableGroup and the value is a List of Variables
+     *          that all belong to the same VariableGroup
+     * @param groupList a list of all previously existing PopulatedDisplayGroup
+     * @param patient the patient to add reference information from
+     */
+    public static void addRefInfo(final Map<VariableGroup, List<Variable>> map, final List<PopulatedDisplayGroup> groupList,
```

Comment:
The parameters for this method seem awkward to me. Why do you need the `Map<VariableGroup, List<Variable>>`? Is it just to tell if there is a `PopulatedDisplayGroup` for that `VariableGroup` already? If so, why not just check `groupList`?